src/os_posix/os_fs.c

```diff
@@ -698,7 +697,8 @@ __posix_file_write_mmap(
     if (pfh->mmap_buf != NULL && !pfh->mmap_resizing && pfh->mmap_size < offset + (wt_off_t)len)
         /* If we are actively extending the file, don't remap it on every write. */
         if ((remap_opportunities++) % WT_REMAP_SKIP == 0) {
-            __wt_prepare_remap_resize_file(file_handle, wt_session);
+            __wt_prepare_remap_resize_file(file_handle, wt_session, offset + (wt_off_t)len, &remap);
```

Comment:
Alternatively here, I could make the prepare function take a NULL boolean and get rid of the new local var here.